src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,127 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { Query } from '../../types/Query';
+
+interface Props {
+  todos: Todo[] | null;
+  setTodos: (todos: Todo[]) => void;
+}
+
+const getSortedTodos = (
+  todos: Todo[] | null,
+  query?: Query,
+  text?: string,
+): Todo[] => {
+  if (!todos) {
+    return [];
+  }
+
+  let newTodos = [...todos];
+
+  if (query) {
+    switch (query) {
+      case 'active': {
+        newTodos = newTodos.filter(todo => !todo.completed);
+
+        break;
+      }
+
+      case 'completed': {
+        newTodos = newTodos.filter(todo => todo.completed);
+
+        break;
+      }
+
+      default:
+        newTodos = newTodos.sort(
+          (todo1, todo2) => todo1.id - todo2.id,
+        );
+
+        break;
+    }
+  }
+
+  if (text) {
+    newTodos = newTodos
+      .filter(todo => todo.title.toLowerCase().includes(text.toLowerCase()));
+  }
+
+  return newTodos;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  todos,
+  setTodos = () => { },
+}) => {
+  const [input, setInput] = useState('');
```

Comment:
Add generic types everywhere for state
```suggestion
  const [input, setInput] = useState<string>('');
```